pyprof/nvtx/nvmarker.py

```diff
@@ -202,9 +198,9 @@ def wrapper_func(*args, **kwargs):
 
             if config.capture_input_ops:
                 ## Stack for callids to work with nested monkey patch function calls
-                patch_list.append(dlprof.call_id)
-                dprint("-----Calling patched function {} call_id {} from class {} len args {} len kwargs {}"\
-                    .format(fn_name, dlprof.call_id, mod, len(args), len(kwargs)))
+                dlprof.patch_list.append(dlprof.call_id)
+                dprint(f"-----Calling patched function {fn_name} call_id {dlprof.call_id} from"\
+                        f" class {mod} len args {len(args)} len kwargs {len(kwargs)}")
```

Comment:
1. The `wrapper_func` is in the critical path. The way the function `dprint` is defined and called here will cause a very expensive string formatting operation irrespective of whether the string is printed or not. Given that in most cases debug is False, the formatted string will end up not being printed. This will slow down profiling significantly.
2. A lot of activity happens inside the function `wrapper_func` and so it would be nice to move such code to a helper function.

Reply:
I just removed the dprints.

Are you saying the *arg parameter that is passed to dprint is evaluated every time regardless of whether debug is set?   I have not noticed a performance issue with this.